ExampleModels/Arithmetic/src/arithmetic.jl

```diff
@@ -0,0 +1,107 @@
+module Arithmetic
+
+abstract type Expression end
+
+struct Constant <: Expression
+    value::Number
+end
+
+@enum OP begin
+    ADD = 1
+    SUB = 2
+    MUL = 3
+    DIV = 4
+end
+
+struct BinaryOp <: Expression
+    left::Expression
+    op::OP
+    right::Expression
+end
+
+function evaluate(expr::Constant)
+    expr.value
+end
+
+function evaluate(expr::BinaryOp) 
+    leftVal = evaluate(expr.left)
```

Comment:
In Julia by convention we use camelCase only for types.  Use `leftval` or `left_val`